### PR TITLE
D1: Prepare to use LiveData instead of Draft model for UI

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #1782

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt [hunk 1]
```diff
@@ -158,53 +159,50 @@ class NewMessageViewModel @Inject constructor(
...
-            true
+            return@runCatching newDraft
         }.getOrElse {
             Sentry.captureException(it)
-            false
+            return@getOrElse null
         }
 
-        if (isSuccess) {
+        if (draft != null) {
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt:185]: `draft?.let`?

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt [hunk 2]
```diff
@@ -158,53 +159,50 @@ class NewMessageViewModel @Inject constructor(
...
-                otherFieldsAreAllEmpty.postValue(false)
-                initializeFieldsAsOpen.postValue(true)
-            }
+            draftInRAM = draft
 
-            initResult.postValue(draftInRAM to signatures)
+            initResult.postValue(InitResult(draft, signatures))
         }
 
+        val isSuccess = draft != null
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt:199]: `emit(draft != null)`?

**Review by tevincent — CHANGES_REQUESTED** [review]

**Review by LunarX — CHANGES_REQUESTED** [review]
Also, what happens when the draft remote id changes compared to the one we keep in the UI

**Review by tevincent — APPROVED** [review]

**Review by FabianDevel — CHANGES_REQUESTED** [review]

**Review by FabianDevel — CHANGES_REQUESTED** [review]
We shpould save the draftRessource in savedStateHandle, because if the user put the app in background before we put the fetched draft in realm, and the system kill the app, then we won't be able to fetch the draft anymore as the resource will be null

**Pull request merged**